docs/Cluster Guides/Scheduling_a_Job.md

```diff
@@ -0,0 +1,108 @@
+
+The equivalent command for GPUs is <code>#SBATCH --gres=gpu:(number of GPUs requested)</code>. This parameter can be set to a maximum of two on R2, and a maximum of one on Borah.
+
+## **Naming, Timing, and Output**
+
+The last three standard parameters you’ll need are <code>#SBATCH -t DD-HH:MM:SS, #SBATCH -o (output file name), and #SBATCH -J (job name)</code>. The first of these is mandatory and tells Slurm how much time you’re requesting for your job. This is formatted as DD-HH:MM:SS. So, for example, running a job for one day and twenty hours would be 1-20:00:00, running for twelve hours and fifty minutes would be 12:50:00, and running for a half hour would be 00:30:00. This is the maximum amount of time you expect your job to run; if a job runs for beyond this specified amount, the scheduler will terminate it. If you have a job that looks like it will exceed the amount of time you’ve specified for it here, please reach out to Research Computing so we can add more time to the job to avoid it being cancelled. The second parameter is the optional output file specifier. This will redirect any terminal output that the job produces to a file with the name you put in the parameter; if this parameter is left out, the output will go into a text file named slurm-(job number).out. The last parameter is the mandatory job name that shows up when the Slurm queue is viewed.
+
+One more parameter that warrants mention is <code>#SBATCH --exclusive</code>. Using this parameter will claim the entire compute node (and all its resources) for just your job, stopping other jobs from running on the node. This is useful if you’re actually utilizing all 28 or 48 (on R2 and Borah, respectively) cores on a node and/or a significant portion of the node’s memory, but it can make the job take a little bit longer to get through the queue.
+
+## **What to Run**
+
+The other component of an sbatch script is a set of arbitrary commands that will be run after the parameters are parsed. These can be essentially any shell commands you could normally run, though obviously only some are useful. As an example, here’s a full sbatch script for an MPI job running a Mandelbrot fractal generator across 28 CPU cores on a single CPU node:
+
+
+<code>#!/bin/bash</code>
+  
+<code>#SBATCH -J MPI_TEST # job name</code>
+  
+<code>#SBATCH -o log_slurm.o%j # output and error file name (%j expands to jobID)</code>
+  
+<code>#SBATCH -n 28 # total number of tasks requested</code>
+  
+<code>#SBATCH -N 1 # number of nodes you want to run on</code>
+  
+<code>#SBATCH -p defq # queue (partition)</code>
+  
+<code>#SBATCH -t 12:00:00 # run time (hh:mm:ss) - 12.0 hours in this example.</code>
+
+module load slurm
+
+module load gcc
+
+module load mpich/ge/gcc/64/3.2.1
+
+mpirun -np 28 ./mandelbrot
```

Comment:
```suggestion
\``` bash title="slurm_mandelbrot.sh"
#!/bin/bash
#SBATCH -J MPI_TEST # job name
#SBATCH -o log_slurm.o%j # output and error file name (%j expands to jobID)
#SBATCH -n 28 # total number of tasks requested
#SBATCH -N 1 # number of nodes you want to run on
#SBATCH -p defq # queue (partition)
#SBATCH -t 12:00:00 # run time (hh:mm:ss) - 12.0 hours in this example.

module load slurm
module load gcc
module load mpich/ge/gcc/64/3.2.1

mpirun -np 28 ./mandelbrot
\```
```

We can edit these lines to display the text as a file as shown above. The "\" in front of the backticks is only there to display the backticks.